arshal_default.go

```diff
@@ -207,11 +207,11 @@ func makeStringArshaler(t reflect.Type) *arshaler {
 		}
 
 		if mo.Flags.Get(jsonflags.StringifyBoolsAndStrings) {
-			b, err := jsontext.AppendQuote(nil, s) // only fails for invalid UTF-8
-			q, _ := jsontext.AppendQuote(nil, b)   // cannot fail since b is valid UTF-8
-			if err != nil && !xe.Flags.Get(jsonflags.AllowInvalidUTF8) {
-				return newMarshalErrorBefore(enc, t, err)
+			b, err := jsonwire.AppendQuote(nil, s, &mo.Flags)
+			if err != nil {
+				return newMarshalErrorBefore(enc, t, &jsontext.SyntacticError{Err: err})
 			}
+			q, _ := jsontext.AppendQuote(nil, b) // cannot fail since b is valid UTF-8
 			return enc.WriteValue(q)
```

Comment:
Previously, the escaping for `EscapeForHTML` and `EscapeForJS` was deferred to the natural behavior of `enc.WriteValue`.

However, to replicate existing v1 behavior, we escape on L210 earlier on by passing in `mo.Flags`, which will have the escape flags specified.